typescript/creational/builder/services/NewsService.ts

```diff
@@ -0,0 +1,23 @@
+import {News} from "./NewsAPI";
+
+export abstract class AbstractNewsService {
+    public abstract allNews(): News[]
+
+    public abstract search(title: string): News[]
+
+    public abstract latestByCategory(category: string): News[]
+}
+
+export class NewsService implements AbstractNewsService {
```

Comment:
If this is where people are expected to start their implementation, add a `TODO` here with some explicit instructions on what to do - assuming that someone may want to tackle this exercise outside of our workshop at a later time.

Reply:
This is now implemented in a "messy" way.